gui/src/connectors/redux/reducers/errors.js

```diff
@@ -0,0 +1,34 @@
+import {
+  MATCH_ROUTE,
+  RECEIVE_SESSION_FAIL
+} from 'actions'
+
+const initialState = {
+  messages: [],
+  currentUrl: null
+}
+const SIGN_IN_FAIL_MSG = 'Your email or password are incorrect. Please try again'
+
+export default (state = initialState, action = {}) => {
+  switch (action.type) {
+    case MATCH_ROUTE: {
+      if (state.currentUrl !== action.match.url) {
+        return Object.assign(
+          {},
+          state,
+          {messages: [], currentUrl: action.match.url}
+        )
+      }
+
+      return state
+    }
+    case RECEIVE_SESSION_FAIL:
+      return Object.assign(
+        {},
+        state,
+        {messages: [SIGN_IN_FAIL_MSG]}
```

Comment:
Do we not want the server to give us specific information about the failure? i.e. "Invalid email" or password, etc.
This introduces a bit of redundancy between the client and the server now, by having two different fail messages. Perhaps we want that to have friendlier language which is ok.

Reply:
Currently the server doesn't send error messages that match the designs which I assumed was on purpose. Agreed that it would be nice to have these in sync